src/Compiler/Service/FSharpCheckerResults.fsi

```diff
@@ -41,6 +45,183 @@ type DelayedILModuleReader =
 /// <summary>Unused in this API</summary>
 type public FSharpUnresolvedReferencesSet = internal FSharpUnresolvedReferencesSet of UnresolvedAssemblyReference list
 
+[<NoComparison; CustomEquality>]
+type internal FSharpFileSnapshot =
+    { FileName: string
+      Version: string
+      GetSource: unit -> Task<ISourceText> }
+
+    member internal Key: ICacheKey<string, string>
+
+    interface ICacheKey<string, string>
+
+type internal FSharpFileSnapshotWithSource =
+    { FileName: string
+      SourceHash: string
+      Source: ISourceText
+      IsLastCompiland: bool
+      IsExe: bool }
+
+/// Referenced assembly on disk. Includes last modified time so we know we need to rebuild when it changes.
+type internal ReferenceOnDisk =
+    { Path: string; LastModified: DateTime }
+
+type internal ProjectSnapshotKey = string * string
+
+[<NoComparison>]
+type internal FSharpProjectSnapshot =
+    {
+        // Note that this may not reduce to just the project directory, because there may be two projects in the same directory.
+        ProjectFileName: string
+
+        /// This is the unique identifier for the project, it is case sensitive. If it's None, will key off of ProjectFileName in our caching.
+        ProjectId: string option
+
+        /// The files in the project.
+        SourceFiles: FSharpFileSnapshot list
+
+        /// Referenced assemblies on disk.
+        ReferencesOnDisk: ReferenceOnDisk list
+
+        /// Additional command line argument options for the project.
+        OtherOptions: string list
+
+        /// The command line arguments for the other projects referenced by this project, indexed by the
+        /// exact text used in the "-r:" reference in FSharpProjectOptions.
+        ReferencedProjects: FSharpReferencedProjectSnapshot list
+
+        /// When true, the typechecking environment is known a priori to be incomplete, for
+        /// example when a .fs file is opened outside of a project. In this case, the number of error
+        /// messages reported is reduced.
+        IsIncompleteTypeCheckEnvironment: bool
+
+        /// When true, use the reference resolution rules for scripts rather than the rules for compiler.
+        UseScriptResolutionRules: bool
+
+        /// Timestamp of project/script load, used to differentiate between different instances of a project load.
+        /// This ensures that a complete reload of the project or script type checking
+        /// context occurs on project or script unload/reload.
+        LoadTime: DateTime
```

Comment:
In CI / test scenarios the resolution of system clock can be too low, leading to some heisenbugs:
https://stackoverflow.com/questions/2143140/c-sharp-datetime-now-precision
I can't recall where was it exactly, but I remember running into this problem at some point in the IncrementalBuild refactor.

Reply:
Good job I'm not really using this then 😄. 

But thanks for the heads up, I'll make sure not to rely on that anywhere.